test/gtest/common/mem_buffer.cc

```diff
@@ -118,6 +118,8 @@ void mem_buffer::set_device_context()
 #if HAVE_CUDA
     if (is_cuda_supported()) {
         cudaSetDevice(0);
+	/* need to call free as context maybe lazily initialized */
```

Comment:
does it trigger context initialization?

Reply:
`cudaSetDevice(0)` may not but `cudaFree(0)` does.